apps/builder/app/builder/features/settings-panel/variables.tsx

```diff
@@ -0,0 +1,362 @@
+import { useId, useMemo, useState } from "react";
+import { nanoid } from "nanoid";
+import { computed } from "nanostores";
+import { useStore } from "@nanostores/react";
+import {
+  Button,
+  CssValueListArrowFocus,
+  CssValueListItem,
+  Flex,
+  FloatingPanelPopoverTitle,
+  InputErrorsTooltip,
+  InputField,
+  Label,
+  ScrollArea,
+  SmallIconButton,
+  Text,
+  Tooltip,
+  theme,
+} from "@webstudio-is/design-system";
+import { MenuIcon, MinusIcon, PlusIcon } from "@webstudio-is/icons";
+import type { DataSource } from "@webstudio-is/sdk";
+import { validateExpression } from "@webstudio-is/react-sdk";
+import {
+  dataSourcesStore,
+  selectedInstanceSelectorStore,
+} from "~/shared/nano-states";
+import { serverSyncStore } from "~/shared/sync";
+import { CodeEditor } from "./code-editor";
+
+type VariableDataSource = Extract<DataSource, { type: "variable" }>;
+
+// convert value expression to js value
+// validating out accessing any identifier
+const parseVariableValue = (code: string) => {
+  const result: { value?: unknown; error?: string } = {};
+  const ids = new Set<string>();
+  try {
+    validateExpression(code, {
+      transformIdentifier: (id) => {
+        ids.add(id);
+        return id;
+      },
+    });
+  } catch (error) {
+    result.error = (error as Error).message;
+    return result;
+  }
+  if (ids.size === 0) {
+    try {
+      result.value = eval(code);
```

Comment:
will not work inside cloudlfare?

Reply:
It's client side code. Variables panel evaluate everything from input

Reply:
Practically json with less verbose syntax